src/app/register/register.component.ts

```diff
@@ -9,6 +10,20 @@ import { FormControl, FormGroup, Validators } from '@angular/forms';
 export class RegisterComponent {
   registerForm = new FormGroup({
     userName: new FormControl('', Validators.required),
-    password: new FormControl('', Validators.required)
+    password: new FormControl('', Validators.required),
   });
+
+  constructor(private registerService: RegisterApiService) {}
+
+  onSubmit() {
+    if (this.registerForm.valid) {
+      const userName = this.registerForm.value.userName as string;
+      const password = this.registerForm.value.password as string;
+      const role = 2 as number;
+
+      this.registerService
```

Comment:
As I already said, this can be in login service. There's no need to create a new service just for register, it can be all together in the login service (`login`, `register`, `logout` etc)